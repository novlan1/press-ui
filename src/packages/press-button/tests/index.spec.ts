import Button from '../press-button.vue';
import { mount } from '@vue/test-utils';

// @ts-ignore
global.wx = {};


test('loading-size prop', () => {
  const wrapper = mount(Button, {
    propsData: {
      loading: true,
      loadingSize: '10px',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('icon-position prop', () => {
  const wrapper = mount(Button, {
    propsData: {
      icon: 'plus',
      iconPosition: 'right',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

// test('click event', () => {
//   const onClick = jest.fn();
//   const wrapper = mount(Button, {
//     context: {
//       on: {
//         click: onClick,
//       },
//     },
//   });

//   wrapper.trigger('click');
//   expect(onClick).toHaveBeenCalled();
// });

// test('not trigger click event when disabled', () => {
//   const onClick = jest.fn();
//   const wrapper = mount(Button, {
//     propsData: {
//       disabled: true,
//     },
//     context: {
//       on: {
//         click: onClick,
//       },
//     },
//   });

//   wrapper.trigger('click');
//   expect(onClick).toHaveBeenCalledTimes(0);
// });

// test('not trigger click event when loading', () => {
//   const onClick = jest.fn();
//   const wrapper = mount(Button, {
//     propsData: {
//       loading: true,
//     },
//     context: {
//       on: {
//         click: onClick,
//       },
//     },
//   });

//   wrapper.trigger('click');
//   expect(onClick).toHaveBeenCalledTimes(0);
// });

// test('touchstart event', () => {
//   const onTouchstart = jest.fn();
//   const wrapper = mount(Button, {
//     context: {
//       on: {
//         touchstart: onTouchstart,
//       },
//     },
//   });

//   wrapper.trigger('touchstart');
//   expect(onTouchstart).toHaveBeenCalled();
// });

test('hide border when color is gradient', () => {
  const wrapper = mount(Button, {
    propsData: {
      color: 'linear-gradient(#000, #fff)',
    },
  });

  // @ts-ignore
  expect(wrapper.element.style.border).toEqual('0px');
});

test('icon-prefix prop', () => {
  const wrapper = mount(Button, {
    propsData: {
      icon: 'success',
      iconPrefix: 'my-icon',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('loading slot', () => {
  const wrapper = mount(Button, {
    propsData: {
      loading: true,
    },
    scopedSlots: {
      loading: () => 'Custom Loading',
    },
  });

  expect(wrapper).toMatchSnapshot();
});
