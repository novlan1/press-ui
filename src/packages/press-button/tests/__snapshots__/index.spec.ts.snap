// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon-position prop 1`] = `
<button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal " iconposition="right"><button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal ">
    <div class="press-button__icon  press-icon-plus press-icon-plus-plus" style="font-size: 1.2em; line-height: inherit;">
      <!---->
      <!---->
    </div>
    <div class="press-button__text"></div>
  </button></button>
`;

exports[`icon-prefix prop 1`] = `
<button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal " iconprefix="my-icon"><button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal ">
    <div class="press-button__icon  press-icon-plus press-icon-plus-success" style="font-size: 1.2em; line-height: inherit;">
      <!---->
      <!---->
    </div>
    <div class="press-button__text"></div>
  </button></button>
`;

exports[`loading slot 1`] = `
<button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal press-button--loading press-button--unclickable "><button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal press-button--loading press-button--unclickable ">
    <div class="press-loading-index">
      <div class=" press-loading">
        <div class="press-loading__spinner press-loading__spinner--circular" style="color: rgb(201, 201, 201); width: 20px; height: 20px;">
          <!---->
        </div>
        <div class="press-loading__text"></div>
      </div>
    </div>
    <!---->
  </button></button>
`;

exports[`loading-size prop 1`] = `
<button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal press-button--loading press-button--unclickable "><button id="" hover-class="press-button--active," lang="" form-type="" open-type="" business-id="" session-from="" send-message-title="" send-message-path="" send-message-img="" show-message-card="" app-parameter="" aria-label="" class=" press-button press-button--default press-button--normal press-button--loading press-button--unclickable ">
    <div class="press-loading-index">
      <div class=" press-loading">
        <div class="press-loading__spinner press-loading__spinner--circular" style="color: rgb(201, 201, 201); width: 10px; height: 10px;">
          <!---->
        </div>
        <div class="press-loading__text"></div>
      </div>
    </div>
    <!---->
  </button></button>
`;
